### ISSUE TITLE
Can't find Incorrect keyword arguments for Enum __init_subclass__ method

### ISSUE THREAD

**Issue by rj-xu** [microsoft/pyright#10851]
**Describe the bug**

Line13, `s=123` should be `str`, incorrect argument.
But if class A is not `Enum`, the error can be found.

**Code or Screenshots**

```python
import enum


class A(enum.IntFlag):
    def __init_subclass__(cls, *, s: str):
        cls.s = s
        return super().__init_subclass__()

    def func(self):
        print(self.s)


class B(A, s=123):
    B1 = enum.auto()
    B2 = enum.auto()


for i in B:
    print(f"{i.name} = {i.value}")
B.B1.func()

```

**VS Code extension or command-line**

ms-python.vscode-pylance 2025.7.1

**Comment by erictraut** [comment]
This is because `IntFlags` derives from `Enum` which has a custom metaclass `EnumMeta` that defines a `__new__` method. When a custom metaclass has a `__new__`, then a type checker can't make assumptions about what that `__new__` method is going to do — and whether it will honor `__init_subclass__`.
